Automattic/jetpack
# Masterbar: Deprecate Jetpack module code

Deprecates the Jetpack Masterbar module code. Same functionality is now provided via the `jetpack-masterbar` package.

## Proposed changes:

* Deprecates the Jetpack Masterbar module code. 
* Updates relevant unit tests

### Other information:


## Jetpack product discussion


pfwV0U-3U-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* **Code Review**
* Use the provided test instructions in https://github.com/Automattic/jetpack/pull/37676 to play around with Masterbar features and confirm no relevant deprecation warnings occur, in a:
1. Self-hosted site
2. WoA classic site
3. WoA default site
4. Simple classic site
5. Simple default site

A reminder on Classic/Default sites: Classic sites are the ones where `wpcom_is_nav_redesign_enabled()` will be `true`.  This means that `return get_blog_option( YOUR_WPCOM_BLOG_ID, 'wpcom_admin_interface' )` returns `wp-admin`

For WoA sites there's a corresponding option to toggle this setting under Settings->General in `wp-admin` for a **classic** WoA site

<img width="727" alt="Screenshot 2024-06-13 at 14 29 19" src="https://github.com/Automattic/jetpack/assets/1758399/0d45e728-f6c8-468d-a397-7d531bd1f79c">

 You can toggle this option via Settings->General in Calypso for a **Default** WoA site:

<img width="781" alt="Screenshot 2024-06-18 at 13 06 21" src="https://github.com/Automattic/jetpack/assets/1758399/28079550-806c-41f3-9937-a688e6529d28">

For Simple ones, you can update the `wpcom_admin_interface` option via `wpsh`
